File: backend/src/scripts/submit_indexnow.ts
import fetch from 'node-fetch';

const API_KEY = "2abfda58abd0480aa668dd43578e1700";
const HOST = "touchflowpro.com";
const KEY_LOCATION = `https://${HOST}/${API_KEY}.txt`;

const coreUrls = [
    '',
    '/free-typing-test',
    '/articles',
    '/about',
    '/faq',
    '/contact',
    '/pricing',
    '/medical-track',
    '/legal-track',
    '/coding-track',
    '/bible-practice'
].map(loc => `https://${HOST}${loc}`);

const SEO_PAGES = [
    'how-to-type-faster',
    'increase-wpm-from-60-to-100',
    'typing-speed-vs-accuracy',
    'how-to-improve-typing-accuracy',
    'typing-speed-averages', 
    'how-to-type-60-wpm',
    'medical-transcription-typing-test',
    'legal-typing-test',
    'typing-test-for-programmers',
    'typing-speed-plateau',
    'improve-typing-speed',
    '60-wpm-to-100-wpm',
    'touchflow-vs-monkeytype',
];

const articleUrls = SEO_PAGES.map(slug => `https://${HOST}/articles/${slug}`);

const allUrls = [...coreUrls, ...articleUrls];

async function submitToIndexNow() {
    console.log(`Submitting ${allUrls.length} URLs to IndexNow for ${HOST}...`);

    const payload = {
        host: HOST,
        key: API_KEY,
        keyLocation: KEY_LOCATION,
        urlList: allUrls
    };

    try {
        const response = await fetch('https://api.indexnow.org/indexnow', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json; charset=utf-8'
            },
            body: JSON.stringify(payload)
        });

        if (response.ok) {
            console.log("✅ Successfully submitted to IndexNow. (HTTP 200 OK)");
        } else if (response.status === 202) {
            console.log("✅ Successfully submitted. IndexNow returned HTTP 202 Accepted (URL validated and added to queue).");
        } else {
            console.error(`❌ IndexNow submission failed with HTTP ${response.status} ${response.statusText}`);
            const text = await response.text();
            console.log("Response body:", text);
        }
    } catch(err) {
        console.error("❌ Error submitting to IndexNow:", err);
    }
}

submitToIndexNow();

File: submit_indexnow.py
import requests
import json

urls = [
    "https://touchflowpro.com/",
    "https://touchflowpro.com/free-typing-test",
    "https://touchflowpro.com/medical-track",
    "https://touchflowpro.com/legal-track",
    "https://touchflowpro.com/coding-track",
    "https://touchflowpro.com/bible-practice",
    "https://touchflowpro.com/about",
    "https://touchflowpro.com/articles",
    "https://touchflowpro.com/contact",
    "https://touchflowpro.com/faq",
    "https://touchflowpro.com/pricing",
    "https://touchflowpro.com/assessment",
    "https://touchflowpro.com/privacy-policy",
    "https://touchflowpro.com/terms",
    "https://touchflowpro.com/articles/touchflow-vs-monkeytype",
    "https://touchflowpro.com/articles/typing-speed-averages",
    "https://touchflowpro.com/articles/ultimate-guide-to-typing-speed",
    "https://touchflowpro.com/articles/how-to-type-faster",
    "https://touchflowpro.com/articles/typing-speed-plateau",
    "https://touchflowpro.com/articles/60-wpm-to-100-wpm",
    "https://touchflowpro.com/articles/type-faster-accurately",
    "https://touchflowpro.com/articles/touch-typing-guide",
    "https://touchflowpro.com/articles/typing-speed-test",
    "https://touchflowpro.com/articles/fastest-typing-techniques",
    "https://touchflowpro.com/articles/typing-practice",
    "https://touchflowpro.com/articles/improve-typing-speed",
    "https://touchflowpro.com/articles/typing-accuracy",
    "https://touchflowpro.com/articles/typing-platform-for-beginners",
    "https://touchflowpro.com/articles/best-typing-platforms-2026",
    "https://touchflowpro.com/articles/typing-speed-for-lawyers",
    "https://touchflowpro.com/articles/icd-10-typing-practice",
    "https://touchflowpro.com/articles/vs-code-typing-hacks"
]

data = {
    "host": "touchflowpro.com",
    "key": "2abfda58abd0480aa668dd43578e1700",
    "keyLocation": "https://touchflowpro.com/2abfda58abd0480aa668dd43578e1700.txt",
    "urlList": urls
}

headers = {
    "Content-Type": "application/json; charset=utf-8"
}

response = requests.post("https://www.bing.com/indexnow", headers=headers, data=json.dumps(data))

print(f"Status Code: {response.status_code}")
print(f"Response: {response.text}")
